features/call/src/main/kotlin/io/element/android/features/call/ui/CallScreenView.kt

```diff
@@ -88,9 +90,20 @@ internal fun CallScreenView(
             },
             onWebViewCreated = { webView ->
                 val interceptor = WebViewWidgetMessageInterceptor(webView)
-                state.eventSink(CallScreeEvents.SetupMessageChannels(interceptor))
+                state.eventSink(CallScreenEvents.SetupMessageChannels(interceptor))
             }
         )
+
+        // We need to restart the Matrix client to get the needed call events
+        if (state.isInWidgetMode) {
```

Comment:
I find it a bit weird to have the view dictate if it should start or stop the sync, could we not manage this entirely in the Presenter?

Reply:
We'd need to use  the `OnLifecycleEvent` there instead, which is possible, but then we need to provide a `LocalLifecycleOwner` in a Molecule test, which is both difficult and kind of weird.

Reply:
Or maybe we could just replace it with a `DisposableEffect`, but that way we don't stop syncing when the call screen goes to background, right?

Reply:
Actually a `DisposableEffect` might be a better solution in the end. I'm replacing it, thanks!